dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,345 @@
+                    throws DotDataException {
 
-    public static String create(String indexName, int shards, boolean live) throws DotIndexException, IOException {
-        if(indexName == null)
-            indexName=ContentletIndexAPIImpl.timestampFormatter.format(new Date());
-        indexName = (live) ? "live_" + indexName : "working_" + indexName;
+        auth(request, response);
 
-        APILocator.getContentletIndexAPI().createContentIndex(indexName, shards);
 
-        return indexName;
-    }
+        final ESIndexAPI esIndexAPI = new ESIndexAPI();
+        final ClusterStats clusterStats = esIndexAPI.getClusterStats();
 
-    public static void activateIndex(String indexName) throws DotDataException {
-    	AdminLogger.log(ESIndexResource.class, "activateIndex", "Trying to activate index: " + indexName);
+        Builder<String, Object> builder =
+                        ImmutableMap.<String, Object>builder().put("clusterName", clusterStats.getClusterName());
 
-        if(indexName.startsWith(SiteSearchAPI.ES_SITE_SEARCH_NAME)){
-            APILocator.getSiteSearchAPI().activateIndex(indexName);
-        }
-        else{
-            APILocator.getContentletIndexAPI().activateIndex(indexName);
+        for (NodeStats stats : clusterStats.getNodeStats()) {
+            builder.put("name", stats.getName())
+                .put("master", stats.isMaster())
+                .put("host", stats.getHost())
+                .put("address", stats.getTransportAddress())
+                .put("size", stats.getSize())
+                .put("count", stats.getDocCount());
         }
-
-        AdminLogger.log(ESIndexResource.class, "activateIndex", "Index activated: " + indexName);
+        return Response.ok(new ResponseEntityView(builder.build())).build();
     }
-
-    public static void deactivateIndex(String indexName) throws DotDataException, IOException {
-    	AdminLogger.log(ESIndexResource.class, "deactivateIndex", "Trying to deactivate index: " + indexName);
-
-        if(indexName.startsWith(SiteSearchAPI.ES_SITE_SEARCH_NAME)){
-            APILocator.getSiteSearchAPI().deactivateIndex(indexName);
-        }
-        else{
-            APILocator.getContentletIndexAPI().deactivateIndex(indexName);
-        }
-
-        AdminLogger.log(ESIndexResource.class, "deactivateIndex", "Index deactivated: " + indexName);
+    
+    @CloseDBIfOpened
+    @GET
+    @JSONP
+    @NoCache
+    @Path("/failed")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response downloadRemainingRecordsAsCsv(@Context final HttpServletRequest request,
+                    @Context final HttpServletResponse response) throws DotDataException {
+        auth(request, response);
+        List<Map<String, Object>> results = new ArrayList<>();
+        
+        
+        
+        APILocator.getReindexQueueAPI().getFailedReindexRecords().stream().forEach(row->{
+            Map<String, Object> failure = new HashMap<>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'failure' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)